File: side-projects/04-obfuscation-playground/examples/js/primes.js
// Sieve of Eratosthenes: primes below 50.
function primesBelow(limit) {
  const sieve = new Array(limit).fill(true);
  sieve[0] = false;
  if (limit > 1) sieve[1] = false;
  for (let i = 2; i < limit; i++) {
    if (sieve[i]) {
      for (let j = i * i; j < limit; j += i) sieve[j] = false;
    }
  }
  const out = [];
  for (let i = 0; i < limit; i++) if (sieve[i]) out.push(i);
  return out;
}

const found = primesBelow(50);
console.log("count", found.length);
console.log("primes", found.join(","));

File: side-projects/04-obfuscation-playground/examples/python/primes.py
# Sieve of Eratosthenes: print the primes below 50.
def primes_below(limit):
    sieve = [True] * limit
    sieve[0] = False
    if limit > 1:
        sieve[1] = False
    for i in range(2, limit):
        if sieve[i]:
            for j in range(i * i, limit, i):
                sieve[j] = False
    return [i for i in range(limit) if sieve[i]]


found = primes_below(50)
print("count", len(found))
print("primes", found)
